### PR TITLE
Add react-a11y to point out accessibility issues

### LINKED ISSUE #244: Use react-a11y
"Warns about potential accessibility issues with your React elements." -- https://github.com/rackt/react-a11y

### PR DESCRIPTION
**Pull request by grapegravity**
Note that this outputs _a lot_ of warnings currently!

Also the stuff about aliasing react-a11y might not be necessary? (But from memory this fixed some prod bundle size issues?)

fixes #244

### CODE REVIEW

**Comment by nelix** [comment]
aliasing is most likely needed because uglify will optimise out only the call to react-a11y but at this point webpack has already added the chunk to the bundle (uglify happens after everything is bundled) and due to the dynamic nature of require uglify can't tell if you will ever require the object.

File: webpack.config.js [hunk 1]
```diff
@@ -2,4 +2,13 @@ var config = require('macropod-tools/webpack.config');
 
 config.module.loaders.push({test: /\.md$/, loaders: ['html-loader','markdown-loader']});
 
+if (typeof config.resolve.alias !== 'object') {
+  config.resolve.alias = {};
+}
+
+if (process.env.NODE_ENV === 'production') {
+  config.resolve.modulesDirectories.push('util');
```
Comment (nelix) [inline webpack.config.js]: config.resolve.alias['react-a11y'] = './utils/noop'; prefered over adding the util folder to the default require path (hard to reason about the side effects of creating filenames in there that match npm modules).

**Comment by ojame** [comment]
i kind of want this to only be run when we pass a flag to npm start (like `a11y`). i think the console is already polluted enough - but, i'm easy.

**Comment by grapegravity** [comment]
I’d almost agree, but I think this, like the linting stuff, needs to be front-and-centre or we’ll just miss it!

**Comment by nelix** [comment]
I would rather have it on by default... The fact that there are lots of warnings is something that really needs adressing outside of this PR

**Comment by grapegravity** [comment]
:+1:

**Comment by ojame** [comment]
i feel like this needs to be addressed proper though - i've heard the word 'accessible' thrown around a lot since i've been here, but no real work done. new components aren't being made accessible, old components aren't being updated. how accessible do we want to be?

if we put this in as default, i feel like we shouldn't merge a PR that touches anything that still raises accessibility warnings - else what is the point?

**Comment by ojame** [comment]
would love to get it to throw - but because there's _many_ existing, it probably won't happen

**Comment by nelix** [comment]
I think thats valid @ojame, we want M-C to be accessible and this step makes it easy for people to start understand what that actually means (although its by no means complete).

This change make it easy to make some small simple fixes to make our basic components useful in more situations.

In general it would be great if working on a module to fix a11y issues first, then continue what you were doing unless they are bug fixes.

**Comment by grapegravity** [comment]
I wholly agree this needs addressing proper. The point of setting this up is to increase our awareness of accessibility issues we might be causing. It’s neither foolproof nor comprehensive, but it’s a start.

While such accessibility has not thus far been a business priority, I’d like to push us to the point where we’re at least consistently considering the implications of the things we’re building.

Additionally, you can easily filter issues by component by going to a components URL and reloading!

**Pull request merged**